src/java/org/apache/cassandra/service/accord/txn/TxnRangeRead.java

```diff
@@ -161,44 +174,54 @@ public AsyncChain<Data> read(Seekable range, SafeCommandStore commandStore, Time
         return AsyncChains.reduce(results, Data::merge);
     }
 
-    private AsyncChain<Data> performLocalRead(PartitionRangeReadCommand command, Range r, int nowInSeconds)
+    public PartitionRangeReadCommand commandForSubrange(Range r, long nowInSeconds)
+    {
+        return commandForSubrange((PartitionRangeReadCommand) get(), r, nowInSeconds);
+    }
+
+    public PartitionRangeReadCommand commandForSubrange(PartitionRangeReadCommand command, Range r, long nowInSeconds)
+    {
+        AbstractBounds<PartitionPosition> bounds = command.dataRange().keyRange();
+        PartitionPosition startPP = bounds.left;
+        PartitionPosition endPP = bounds.right;
+        TokenKey startTokenKey = new TokenKey(command.metadata().id, startPP.getToken());
+        Token subRangeStartToken = ((AccordRoutingKey)r.start()).asTokenKey().token();
+        Token subRangeEndToken = ((AccordRoutingKey)r.end()).asTokenKey().token();
```

Comment:
Yes and it does fail. Need to explicitly check for sentinel key or just stop using it entirely and use `MinTokenKey`.

Reply:
Actually can't use `MinTokenKey` on the right since we don't have the the max token. At least sort of we don't, it was added and used for `AccordRoutingKey.SentinelKey.toTokenKeyBroken` but I am not sure if that is something that should really be used.

Reply:
Yeah it looks like the maximum token isn't actually serializable for OPP.